pkgs/development/compilers/lunarml/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, fetchFromGitHub
+, fetchzip
+, stdenvNoCC
+, mlton
+, lua5_3
+, luajit
+, nodejs
+, withExamples ? false
+}:
+
+let
+  pname = "lunarml";
+in
+stdenvNoCC.mkDerivation {
+  inherit pname;
+
+  version = "unstable-2022-01-24";
+
+  src = fetchFromGitHub {
+    owner = "minoki";
+    repo = "LunarML";
+    rev = "60b646ae7d55550626642a1b3f9a0af3944cc2a6";
+    sha256 = "sha256-lI947OQlb5s9tkSns9bP4uNyjsa1fsPuPiczgfVRe1I=";
+  };
+
+  nativeBuildInputs = [
+    mlton
+  ];
+
+  nativeCheckInputs = [
+    lua5_3
+    luajit
+    nodejs
```

Comment:
AFAICT we're not actually using luajit or nodejs, because we're only building the `test` target; see https://github.com/minoki/LunarML#building. Those should be added via `checkTarget` or `checkFlags`.